legs:
  joint_states_controller:
      type: joint_state_controller/JointStateController
      publish_rate: 50

  FL_hip_joint_position_controller:
      type: effort_controllers/JointPositionController
      joint: FL_hip_joint
      pid: {p: 20, d: 0.1, i: 0}
  FL_thigh_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: FL_thigh_joint
    pid: { p: 20, d: 0.1, i: 0}
  FL_calf_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: FL_calf_joint
    pid: { p: 20, d: 0.1, i: 0}


  FR_hip_joint_position_controller:
      type: effort_controllers/JointPositionController
      joint: FR_hip_joint
      pid: {p: 20, d: 0.1, i: 0}
  FR_thigh_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: FR_thigh_joint
    pid: { p: 20, d: 0.1, i: 0}
  FR_calf_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: FR_calf_joint
    pid: { p: 20, d: 0.1, i: 0}

  
  RL_hip_joint_position_controller:
      type: effort_controllers/JointPositionController
      joint: RL_hip_joint
      pid: {p: 20, d: 0.1, i: 0}
  RL_thigh_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: RL_thigh_joint
    pid: { p: 20, d: 0.1, i: 0}
  RL_calf_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: RL_calf_joint
    pid: { p: 20, d: 0.1, i: 0}
  
  RR_hip_joint_position_controller:
      type: effort_controllers/JointPositionController
      joint: RR_hip_joint
      pid: {p: 20, d: 0.1, i: 0}
  RR_thigh_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: RR_thigh_joint
    pid: { p: 20, d: 0.1, i: 0}
  RR_calf_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: RR_calf_joint
    pid: { p: 20, d: 0.1, i: 0}